### DISCUSSION TITLE
New backend: KubeVirt

### DISCUSSION THREAD

**Discussion by Wouterkoorn** [ContainerSSH/ContainerSSH#566] [Ideas]
Instead of spawning containers with the Kubernetes backend selection, KubeVirt (https://kubevirt.io/) could possibly be utilized to spawn virtual machines instead of containers.

This would not conform the the name **Container**SSH, but I guess that is already the case with the SSH proxy backend.

Thoughts?

**Comment by ghost** [comment]
Hey @Wouterkoorn this is possible today. You can create the KubeVirt VM on the configuration server (needs a bit of code) and you can use the SSH proxy backend to forward the connection into the VM. If this would be a separate backend, it would have to use the SSH proxy backend because AFAIK KubeVirt has no native API for accessing the VM console.

I'd also like to note that KubeVirt doesn't run on AppArmor systems such as Ubuntu and testing it in GitHub Actions would be difficult since KubeVirt requires nested virtualization if I remember correctly, which is not available on GHA.

It's also worth noting that backends are one of the harder parts of ContainerSSH to write and test, thier proliferation isn't necessarily desirable unless there is a strong and compelling use case that cannot be otherwise met.
